gtest/restricted_dep_types.cpp

```diff
@@ -231,17 +236,26 @@ TEST(RestrictedDependentTypes, join_singleton) {
 TEST(RestrictedDependentTypes, ll) {
     World w;
     Normalizers normalizers;
+    Passes passes;
+
+    auto compile_d = Dialect::load("compile", {});
+    compile_d.register_normalizers(normalizers);
+    // compile_d.register_passes(passes);
+    fe::Parser::import_module(w, "compile", {}, &normalizers);
 
     auto mem_d = Dialect::load("mem", {});
     mem_d.register_normalizers(normalizers);
+    // mem_d.register_passes(passes);
     fe::Parser::import_module(w, "mem", {}, &normalizers);
 
     auto core_d = Dialect::load("core", {});
     core_d.register_normalizers(normalizers);
+    // core_d.register_passes(passes);
     fe::Parser::import_module(w, "core", {}, &normalizers);
 
     auto math_d = Dialect::load("math", {});
     math_d.register_normalizers(normalizers);
+    // math_d.register_passes(passes);
     fe::Parser::import_module(w, "math", {}, &normalizers);
```

Comment:
what about the documented code lines? Can we remove them or are they considered to be used in the future?

Reply:
In theory, we should need the passes to be registered for each loaded dialect.
There was an issue at this point. But this is related to default compilations which will be handled in a separate PR.